pkg/attestation/crafter/crafter.go

```diff
@@ -327,7 +325,7 @@ func sanitizeRemoteURL(remoteURL string) (string, error) {
 	return uri.String(), nil
 }
 
-func initialCraftingState(cwd string, schema *schemaapi.CraftingSchema, wf *api.WorkflowMetadata, dryRun bool, runnerType schemaapi.CraftingSchema_Runner_RunnerType, jobURL string, blockOnPolicyViolation bool) (*api.CraftingState, error) {
+func initialCraftingState(cwd string, opts *InitOpts) (*api.CraftingState, error) {
```

Comment:
much cleaner, but now it might make sense to add some properties validations as a safeguard in this method, basically checking for required options